bears/generate_package.py

```diff
@@ -159,12 +161,23 @@ def main():
                                       substitution_dict)
 
             bear_dist_name = bear_name + '-' + bear_version
-            if args.register:
+            if args.register and not args.bear:
                 perform_register(os.path.join('bears', 'upload', bear_name),
                                  bear_dist_name)
-            if args.upload:
+            if args.upload and not args.bear:
                 perform_upload(os.path.join('bears', 'upload', bear_name))
 
+    if args.bear:
```

Comment:
also, shouldnt we have this at the start and just do a "return" "if args.bear" so we wouldnt have to check "if args.register and not args.bear" each time? as in it wouldnt even reach this "if" if args.bear was set.